lib/spack/spack/cmd/buildcache.py

```diff
@@ -207,13 +207,7 @@ def setup_parser(subparser: argparse.ArgumentParser):
 
     # Download tarball and specfile
     download = subparsers.add_parser("download", help=download_fn.__doc__)
-    download_spec_or_specfile = download.add_mutually_exclusive_group(required=True)
```

Comment:
I would do deprecations in a separate PR.

Reply:
I opened #49999 for this, and will remove from here next.